lib/rubocop/cop/rails/has_many_or_has_one_dependent.rb

```diff
@@ -33,11 +33,17 @@ class HasManyOrHasOneDependent < Cop
           (pair (sym :dependent) !(:nil))
         PATTERN
 
+        def_node_matcher :has_through?, <<-PATTERN
+          (pair (sym :through) !(:nil))
```

Comment:
It's possible to do `has_many ..., through: :nil`? If this is an edge case, let's add a test for it. 🙂

Reply:
Extended specs and mentioned `:through` option in the documentation for the cop.

Reply:
@Drenmi  I've checked, `through: nil` is a possible case and Rails will not complain about it. But it will throw an error once you try to use that relation in an application. So I think it makes sense to have this check.